nodejs/week2/app.js

```diff
@@ -0,0 +1,68 @@
+import * as dotenv from "dotenv";
+dotenv.config();
+import express from "express";
+const app = express();
+const port = 3000;
+import doc from "./documents.json" assert { type: "json" };
+
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send("Test message");
+});
+
+app.get("/search", (req, res) => {
+  const { q } = req.query;
+  let result = doc;
+  if (q) {
+    result = doc.filter((obj) =>
+      Object.keys(obj).some((key) =>
+        obj[key].toString().toLowerCase().includes(q.toLowerCase())
```

Comment:
You could consider moving this piece of code as reusable by creating a function.